Update runtime and add gnome-common because it's not in the runtime

lionirdeadman opened the pull request.

lionirdeadman commented:
cc : hfiguiere

lionirdeadman commented:
cc @hfiguiere

hfiguiere commented:
I can merge otherwise.

hadess commented:
> I can merge otherwise.

Consider this a request to not ever do that on one of the modules I maintain. I'd like to be able to choose my own co-maintainers.

hadess commented on appdata.patch:
```diff
@@ -0,0 +1,14 @@
+   </screenshots>
++  <releases>
++    <release version="1.0.3" date="2018-10-24"/>
++  </releases>
```
This change needs to be done upstream.

lionirdeadman replied:
Upstream hasn't made a release in ~3years, I'll send a PR but I think the patch should be kept anyways because otherwise I need to skip the appstream check for flathub to be happy about it.

lionirdeadman replied:
I've made the PR, mark this as resolved if you believe that the patch should be kept here until a next release.

hadess replied:
This is the great thing about the manifest mentioning a specific git commit, you can change it without waiting for a new release. So you'll need 3 separate commits:
- one to change the commit we're pointing to, so it includes the appstream patch
- one to install gnome-common (make sure to include a `"cleanup": "0"` equivalent so all the gnome-common files get removed from the installed application)
- one to bump the SDK version

Let me know if any of those are unclear.

lionirdeadman replied:
Is changing the commit appropriate? Does it only have translation and this appdata in difference with 1.0.3?

Also, my change from commit to tag is simply because 1.0.3 is not a commit so it shouldn't be specified as such. I've changed it to specify the tag and commit as I've told is the appropriate thing to do.

I'm fine modifying only the commit to specify this appstream change if you can confirm that it doesn't change application code.

As for how I've cut the commits, I've already started cutting it a bit ago. I guess I'll redo it to include those changes.

lionirdeadman replied:
I checked the commits, expected more of them but it ended up being rather short. There's also translation changes.

hadess commented on org.gnome.OfficeRunner.json:
```diff
@@ -17,13 +17,23 @@
                     "type": "git",
                     "url": "https://gitlab.gnome.org/GNOME/office-runner.git",
-                    "commit": "1.0.3"
+                    "tag": "1.0.3"
```
Why does this need to be changed?

lionirdeadman replied:
1.0.3 is not a commit, that'd be invalid, no? I'm not sure how it worked otherwise.

hadess replied:
If it worked and now doesn't, then the change that causes that needs to be referenced. (Or fixed if it's a bug)

hadess replied:
Mark this as resolved when you bump the commit to use to include the appstream changes.

hadess commented on org.gnome.OfficeRunner.json:
```diff
@@ -17,13 +17,23 @@
     ],
     "modules": [
+        {
+            "name": "gnome-common",
```
Please split the gnome-common addition from the runtime-version bump.

lionirdeadman replied:
It does not build without it, do you mean two different commits?

hadess replied:
Yes, two different commits. You can install gnome-common even if it's already installed in the runtime.

Commit:
Update runtime

This updates the runtime and SDK of the application to use
GNOME 40 rather than the EOL GNOME 3.32

Commit:
Add gnome-common

As per https://gitlab.gnome.org/GNOME/gnome-build-meta/-/issues/113
gnome-common was removed from the GNOME runtime since 3.34

Commit:
Change commit to the one adding appstream release data

This also includes translations and no code changes since 1.0.3
therefore it is acceptable to do

lionirdeadman commented:
I believe I've done all the changes requested so I've marked them as resolved. Let me know if there's anything else to change.

This was good git practice, haha.

hadess commented:
> This was good git practice, haha.

Thanks for the patches!
